chainer/functions/noise/zoneout.py

```diff
@@ -0,0 +1,58 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class Zoneout(function.Function):
+
+    """Zonepout regularization."""
+
+    def __init__(self, zoneout_ratio):
+        self.zoneout_ratio = zoneout_ratio
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+
+    def forward(self, inputs):
+        h, x = inputs
+        xp = cuda.get_array_module(*x)
+        if xp == numpy:
+            flag_x = xp.random.rand(*h[0].shape) >= self.zoneout_ratio
+        else:
+            flag_x = (xp.random.rand(*h[0].shape, dtype=numpy.float32) >=
```

Comment:
What is the reason why you use float32 instead of float64(default)? I am wondering if we need it as we convert it into boolean array anyway.